redis/src/main/java/org/hibernate/ogm/datastore/redis/options/impl/AssociationStorageOption.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.redis.options.impl;
+
+import org.hibernate.ogm.datastore.document.cfg.DocumentStoreProperties;
+import org.hibernate.ogm.datastore.document.options.AssociationStorageType;
+import org.hibernate.ogm.options.spi.UniqueOption;
+import org.hibernate.ogm.util.configurationreader.spi.ConfigurationPropertyReader;
+
+/**
+ * Represents the type of association storage as configured via the API or annotations for a given element.
+ *
+ * @author Gunnar Morling
+ * @author Mark Paluch
+ */
+public class AssociationStorageOption extends UniqueOption<AssociationStorageType> {
```

Comment:
Ok from my side but that's up to you. `AssociationStorageOption` has a `document` in its package name, was not sure about that and wanted to see, what's your opinion about that. Will change it.